urlcaef331f4e_dreamable_cra_tgz-pJ8-coho_lp_project_LPProjectJ8/src/coho/interp/Parser.java

```diff
@@ -0,0 +1,585 @@
+    "\012\015\016\004\001\002\000\004\015\ufffc\001\002\000" +
+    "\012\011\ufff3\013\ufff3\014\ufff3\015\ufff3\001\002" });
+
+  /** Access to parse-action table. */
+  public short[][] action_table() {return _action_table;}
+
+  /** <code>reduce_goto</code> table. */
+  protected static final short[][] _reduce_table = 
+    unpackFromStrings(new String[] {
+    "\000\051\000\012\003\011\004\005\005\013\012\006\001" +
+    "\001\000\004\005\052\001\001\000\002\001\001\000\002" +
+    "\001\001\000\010\003\046\004\005\005\013\001\001\000" +
+    "\002\001\001\000\002\001\001\000\002\001\001\000\002" +
+    "\001\001\000\002\001\001\000\010\005\033\006\034\010" +
+    "\032\001\001\000\004\005\017\001\001\000\002\001\001" +
+    "\000\002\001\001\000\002\001\001\000\006\005\022\006" +
+    "\023\001\001\000\002\001\001\000\002\001\001\000\002" +
+    "\001\001\000\006\005\027\007\026\001\001\000\002\001" +
+    "\001\000\002\001\001\000\006\005\027\007\031\001\001" +
+    "\000\002\001\001\000\002\001\001\000\002\001\001\000" +
+    "\002\001\001\000\002\001\001\000\006\005\040\011\037" +
+    "\001\001\000\002\001\001\000\002\001\001\000\006\005" +
+    "\040\011\042\001\001\000\002\001\001\000\002\001\001" +
+    "\000\002\001\001\000\002\001\001\000\002\001\001\000" +
+    "\002\001\001\000\004\005\051\001\001\000\002\001\001" +
+    "\000\002\001\001" });
+
+  /** Access to <code>reduce_goto</code> table. */
+  public short[][] reduce_table() {return _reduce_table;}
+
+  /** Instance of action encapsulation class. */
+  protected CUP$Parser$actions action_obj;
+
+  /** Action encapsulation object initializer. */
+  protected void init_actions()
+    {
+      action_obj = new CUP$Parser$actions(this);
+    }
+
+  /** Invoke a user supplied parse action. */
+  public java_cup.runtime.Symbol do_action(
+    int                        act_num,
+    java_cup.runtime.lr_parser parser,
+    java.util.Stack            stack,
+    int                        top)
+    throws java.lang.Exception
+  {
+    /* call code in generated class */
+    return action_obj.CUP$Parser$do_action(act_num, parser, stack, top);
+  }
+
+  /** Indicates start state. */
+  public int start_state() {return 0;}
+  /** Indicates start production. */
+  public int start_production() {return 0;}
+
+  /** <code>EOF</code> Symbol index. */
+  public int EOF_sym() {return 0;}
+
+  /** <code>error</code> Symbol index. */
+  public int error_sym() {return 1;}
+
+
+  /** Scan to get the next Symbol. */
+  public java_cup.runtime.Symbol scan()
+    throws java.lang.Exception
+    {
+
+  throw new RuntimeException("this method needs to be overridden");
+
+    }
+
+
+  protected void init_actions_2(PTfactory pf, BlockingQueue<PTeval> bq)
+  { action_obj = new CUP$Parser$actions(this,pf,bq); }
+
+}
+
+/** Cup generated class to encapsulate user supplied action code.*/
+class CUP$Parser$actions {
+
+
+  void enqueue(PTeval nd) {
+	try{
+		bq.put(nd);
+	}catch(InterruptedException e){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)